sdk/communication/Azure.Communication.MediaComposition/src/MediaCompositionClient.cs

```diff
@@ -0,0 +1,320 @@
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Create");
+            scope.Start();
+            try
+            {
+                // TODO: what is this id and is it needed?
+                var id = new Guid().ToString();
+                return await RestClient.CreateAsync(mediaCompositionId, id, layout, inputs, outputs, CompositionStreamState.NotStarted, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id. </param>
+        /// <param name="layout"> The layout to compose the media. </param>
+        /// <param name="inputs"> The media inputs. </param>
+        /// <param name="outputs"> The media outputs. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual Response<MediaCompositionBody> Create(string mediaCompositionId, MediaCompositionLayout layout, IDictionary<string, MediaInput> inputs, IDictionary<string, MediaOutput> outputs, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Create");
+            scope.Start();
+            try
+            {
+                // TODO: what is this id and is it needed?
+                var id = new Guid().ToString();
+                return RestClient.Create(mediaCompositionId, id, layout, inputs, outputs, CompositionStreamState.NotStarted, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <param name="mediaCompositionId"> The media composition id of the composition to update. </param>
+        /// <param name="layout"> The updated layout to compose media. </param>
+        /// <param name="inputs"> The updated media inputs. </param>
+        /// <param name="outputs"> The updated media outputs. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public virtual async Task<Response<MediaCompositionBody>> UpdateAsync(string mediaCompositionId, MediaCompositionLayout layout = null, IDictionary<string, MediaInput> inputs = null, IDictionary<string, MediaOutput> outputs = null, CancellationToken cancellationToken = default)
+        {
+            using var scope = _clientDiagnostics.CreateScope("MediaCompositionClient.Update");
+            scope.Start();
+            try
+            {
+                // TODO: what is this id and how to retrieve for updates?
+                // Should the CompositionStreamState be exposed for update?
```

Comment:
Commenting for visibility:

- What is this id and how to retrieve for updates?
- Should the CompositionStreamState be exposed for update?

Reply:
Had a discussion with team, the CompositionStreamState will be updated to readOnly and should not be exposed to set